lib/apimatic-core/utilities/date_time_helper.rb

```diff
@@ -116,7 +116,7 @@ def self.from_unix(date_time)
     # @return [DateTime] A DateTime object.
     def self.from_rfc3339(date_time)
       # missing timezone information
-      if date_time.end_with?('Z') || date_time.index('+')
+      if date_time.end_with?('Z') || date_time.index('+') || date_time.index('-')
         DateTime.rfc3339(date_time)
       else
         DateTime.rfc3339("#{date_time}Z")
```

Comment:
@sufyankhanrao can you also add a unit test for missing timezone information i.e datetime without 'Z'? I believe the check for the index of '-' is always true for any datetime. So the else block would never execute

Reply:
yes you are right, let me fix it.

Reply:
Done

Reply:
Can you use the regex `/T.*[+-]|T.*Z/`? your current regex only supports [+-] hh:mm, but `rfc3339` standard also supports [+-] hh:mm:ss as well as ( [+-] hhmm and [+-] hhmmss ) without colons

Reply:
Another issue here is that the `rfc3339` format also supports just dates like `2018-01-01` in such cases you shouldn't append Z after that date